test_cli.py

```diff
@@ -126,10 +162,25 @@ def run_command(
             kwargs = dict(kwargs, encoding="utf-8")
         if include_stderr:
             kwargs = dict(kwargs, stderr=subprocess.STDOUT)
-        output = subprocess.check_output(
-            [a.encode() if type(a) is str else a for a in args], **kwargs
-        )
-        exitcode = 0
+        if not force_subprocess and isinstance(args, list) and args[0] == cl:
+            # In this case, run the codalab CLI directly, which is much faster
+            # than opening a new subprocess to do so.
+            # We skip doing this if force_subprocess is set to true (which forces
+            # us to use subprocess even for cl commands.)
+            _ = io.StringIO()  # Not used; we just don't want to redirect cli.stderr to f.
+            f = FakeStdout()
+            cli = BundleCLI(CodaLabManager(), stdout=f, stderr=_)
```

Comment:
Might not be great to have stderr just disappear?  Why not redirect to `f`?

Reply:
Actually, that's how it worked before. Our call to `subprocess.check_output` discarded stderr, except for the case in which `include_stderr=True` is passed. That is passed only in the test module `gen-readthedocs` (https://github.com/codalab/codalab-worksheets/blob/4d0ef5b009dc4420559d6803b3b6ddd3a029d3b9/test_cli.py#L589) which will use the subprocess anyway.

If you redirect stderr to `f`, it fails several other tests because they currently don't expect to see stderr input suddenly coming into stdout.